src/org/openbravo/erpCommon/utility/reporting/printing/PrintController.java

```diff
@@ -497,6 +508,91 @@ protected void post(HttpServletRequest request, HttpServletResponse response,
+   *     the report associated with the document
+   * @throws JSONException
+   *     if there is an error processing the JSON parameters
+   * @throws IOException
+   *     if there is an error retrieving the report file path
+   */
+  private static void setPostHookParams(DocumentType documentType, JSONObject jsonParams, String documentId,
+      Report report) throws JSONException, IOException {
+    jsonParams.put(DOCUMENT_ID, documentId);
+    jsonParams.put(DOCUMENT_TYPE, documentType);
+    jsonParams.put(REPORT_FILE_PATH, report.getTargetLocation());
+  }
+
+  /**
+   * Sets the parameters for preProcess hooks in the given JSON parameters.
+   *
+   * @param documentType
+   *     the type of document being processed
+   * @param jsonParams
+   *     the JSON parameters being processed
+   * @param documentId
+   *     the ID of the document being processed
+   * @throws JSONException
+   *     if there is an error processing the JSON parameters
+   */
+  private static void setPreHookParams(DocumentType documentType, JSONObject jsonParams,
+      String documentId) throws JSONException {
+    jsonParams.put(DOCUMENT_ID, documentId);
+    jsonParams.put(DOCUMENT_TYPE, documentType);
+  }
+
+  /**
+   * Executes the preProcess hooks for the given document type and handles any errors encountered during execution.
+   *
+   * @param documentType
+   *     the type of document being processed
+   * @param hookManager
+   *     the manager responsible for handling hooks
+   * @param jsonParams
+   *     the JSON parameters being processed
+   * @throws JSONException
+   *     if there is an error processing the JSON parameters
+   * @throws OBException
+   *     if an error occurs during the execution of hooks
+   */
+  private static void executePreProcessHooks(DocumentType documentType, PrintControllerHookManager hookManager,
+      JSONObject jsonParams) throws JSONException {
+    try {
+      hookManager.executeHooks(jsonParams, hookManager.getPreProcess());
+    } catch (final OBException e) {
+      String documentNo = getDocumentNo(documentType, jsonParams);
+      throw new OBException(String.format(OBMessageUtils.messageBD("Error_Printing_Document"),
+          "<li>" + documentNo + ": " + e.getMessage() + "</li>"));
+    }
+  }
+
+  /**
+   * Retrieves the document number for a specified document type and document ID.
+   * The document must be active (isactive = 'Y').
+   *
+   * @param documentType
+   *     the type of the document for which to retrieve the document number
+   * @param jsonParams
+   *     a JSONObject containing the parameters, including the document ID
+   * @return the document number as a String
+   * @throws JSONException
+   *     if there is an error parsing the document ID from jsonParams
+   */
+  private static String getDocumentNo(DocumentType documentType, JSONObject jsonParams) throws JSONException {
+    String query = String.format("SELECT documentNo FROM %s WHERE %s_id = :id AND isactive = 'Y'",
```

Comment:
You must get this information of sequence functions or their table. Also the same table can be used on different documents types, ie: C_Order table can be used to sales or purchase document types.